src/frontend/src/catalog/system_catalog/rw_catalog/iceberg_tables.rs

```diff
@@ -0,0 +1,155 @@
+use thiserror_ext::AsReport;
+
+use crate::catalog::root_catalog::SchemaPath;
+use crate::catalog::system_catalog::SysCatalogReaderImpl;
+use crate::error::{ErrorCode, Result};
+
+// JDBC/SQL catalog integration docs: https://iceberg.apache.org/docs/latest/jdbc/#configurations
+// `iceberg_tables` definition in iceberg java sdk https://github.com/apache/iceberg/blob/4850b622c778deb4b234880bfd7643070e0a5458/core/src/main/java/org/apache/iceberg/jdbc/JdbcUtil.java#L125-L146
+// This system table is used to store the iceberg tables' metadata and only show the tables that the user has access to,
+// so it can be used by other query engine to fetch iceberg catalog and provide an access control layer.
+
+#[derive(Fields)]
+#[primary_key(catalog_name, table_namespace, table_name)]
+struct IcebergTables {
+    pub catalog_name: String,
+    pub table_namespace: String,
+    pub table_name: String,
+    pub metadata_location: Option<String>,
+    pub previous_metadata_location: Option<String>,
+    pub iceberg_type: Option<String>,
+}
+
+#[system_catalog(table, "rw_catalog.iceberg_tables")]
+async fn read(reader: &SysCatalogReaderImpl) -> Result<Vec<IcebergTables>> {
+    let meta_store_endpoint = reader.meta_client.get_meta_store_endpoint().await?;
+
+    let meta_store_endpoint = url::Url::parse(&meta_store_endpoint).map_err(|_| {
+        ErrorCode::InternalError("failed to parse the meta store endpoint".to_owned())
+    })?;
+    let meta_store_backend = meta_store_endpoint.scheme().to_owned();
+    let meta_store_user = meta_store_endpoint.username().to_owned();
+    let meta_store_password = meta_store_endpoint
+        .password()
+        .ok_or_else(|| {
+            ErrorCode::InternalError("failed to parse password from meta store endpoint".to_owned())
+        })?
+        .to_owned();
+    let meta_store_host = meta_store_endpoint
+        .host_str()
+        .ok_or_else(|| {
+            ErrorCode::InternalError("failed to parse host from meta store endpoint".to_owned())
+        })?
+        .to_owned();
+    let meta_store_port = meta_store_endpoint.port().ok_or_else(|| {
+        ErrorCode::InternalError("failed to parse port from meta store endpoint".to_owned())
+    })?;
+    let meta_store_database = meta_store_endpoint
+        .path()
+        .trim_start_matches('/')
+        .to_owned();
+
+    let Ok(meta_backend) = MetaBackend::from_str(&meta_store_backend, true) else {
+        bail!("failed to parse meta backend: {}", meta_store_backend);
+    };
+
+    if !matches!(meta_backend, MetaBackend::Postgres) {
+        bail!("only support `iceberg_tables` in postgres backend");
+    }
+
+    let conn_str = format!(
+        "host={} port={} user={} password={} dbname={}",
+        meta_store_host, meta_store_port, meta_store_user, meta_store_password, meta_store_database
+    );
+
+    let (client, conn) = tokio_postgres::connect(&conn_str, tokio_postgres::NoTls)
+        .await
+        .map_err(|e| anyhow!(e))?;
```

Comment:
Hmmm, this looks concerning. Does frontend pod have access to the meta backend RDS? (should it have?)

Reply:
Maybe having a meta RPC to do the query is preferred.

Reply:
I just refactor this part and use a meta RPC instead